test-ng startup time check fails due to slow user data script on azure

On azure the test-ng framework regularly fails with

```
FAILED test_basics.py::test_startup_time - AssertionError: Userspace startup too slow: 68.6s (tolerated 60.0s)
```

`systemd-analyze` reveals this is due to `cloud-final.service` taking `40.826s` (exact times obviously differ each run). Looking at the cloud init logs:

```
2025-12-15 11:11:05,758 - subp.py[DEBUG]: Running command ['/var/lib/cloud/instance/scripts/part-001'] with allowed return codes [0] (shell=False, capture=False)
2025-12-15 11:11:46,315 - performance.py[DEBUG]: Running ['/var/lib/cloud/instance/scripts/part-001'] took 40.557 seconds
```

So running the user-data script is what causes the delay. Most likely the attachment of the test-ng disk is what is causing the slow down here, as this is done in a loop with a 10 sec sleep after each attempt:

https://github.com/gardenlinux/gardenlinux/blob/3ddc8cfb1841920e58685671094266882259e32f/tests-ng/util/run_cloud.sh#L159-L163

This means the test is not accurately measuring the system startup time itself, but is also measuring its own drive attachment. This will need to be adjusted, e.g. by moving the drive attachment logic to be performed later over the ssh connection, NOT directly in the user data script. This script should probably exclusively enable the ssh server.

--- a/tests-ng/util/run_cloud.sh
+++ b/tests-ng/util/run_cloud.sh
@@ -150,22 +150,31 @@ if [ ! -f "$ssh_private_key" ]; then
 fi
 
 user_data_script="$(mktemp)"
-cat >"$user_data_script" <<EOF
+cat >"$user_data_script" <<'EOF'
 #!/usr/bin/env bash
 
 systemctl enable --now ssh
 
 mkdir /run/gardenlinux-tests
+cat > /run/gardenlinux-tests/run_tests << 'RUNTESTSEOF'
+#!/usr/bin/env bash
+
 # disk attachment might take a while
-for i in \$(seq 1 12); do
+for i in $(seq 1 12); do
 	mount /dev/disk/by-label/GL_TESTS /run/gardenlinux-tests && break
 	sleep 10
 done
 
-if ! mountpoint /run/gardenlinux-tests; then
+cd "$PWD"
+
+if ! mountpoint /run/gardenlinux-tests > /dev/null; then
 	exit 1
 fi
 mkdir -p /run/gardenlinux-tests/tests/log
+
+exec "$0" "$@"
+RUNTESTSEOF
+chmod +x /run/gardenlinux-tests/run_tests
 EOF
 
 if ((cloud_image)); then

